docs/site/Loopback-component-authorization.md

```diff
@@ -0,0 +1,346 @@
+---
+lang: en
+title: 'Authorization'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Loopback-component-authorization.html
+---
+
+## Overview
+
+In every web application, we need to have a way to identify access rights of a
+user for any resource, which is known as **Authorization**. This is a
+minimalistic guide for creating such an implementation using Loopback component.
+This can be part of your main REST Application project or can be created as a
+Loopback extension for reuse in multiple projects. Latter is the better option
+for obvious reasons - reusability.
+
+## The requirement
+
+1. Every protected API end point needs to be restricted by specific permissions.
+2. API allows access only if logged in user has permission as per end point
+   restrictions.
+3. API throws **403 Forbidden** error if logged in user do not have sufficient
+   permissions.
+4. Publicly accessible APIs must be accessible regardless of user permissions.
+5. Every user has a set of permissions. These permissions may be associated via
+   role attached to the user or directly to the user.
+
+## The implementation
+
+First, let's define the types needed for this.
+
+{% include code-caption.html content="/src/authorization/types.ts" %}
+
+```ts
+/**
+ * Authorize action method interface
+ */
+export interface AuthorizeFn {
+  (userPermissions?: string[]): Promise<boolean>;
```

Comment:
Can we describe `userPermissions`? Does it represent the granted permissions or mapped roles for the user?

Reply:
Description added.